src/main/java/org/icgc/argo/program_service/services/ProgramService.java

```diff
@@ -302,16 +302,8 @@ public void removeProgram(String name) throws EmptyResultDataAccessException {
   }
 
   public List<ProgramEntity> listPrograms() {
-    val programs =
-        programRepository.findAll(
-            new ProgramSpecificationBuilder()
-                .setFetchCancers(true)
-                .setFetchPrimarySites(true)
-                .setFetchInstitutions(true)
-                .setFetchCountries(true)
-                .setFetchRegions(true)
-                .listAll());
-    return List.copyOf(programs);
+    val programs = programRepository.findAll();
+    return programs.stream().map(program -> getProgram(program.getShortName())).collect(Collectors.toList());
```

Comment:
what is the point of this line? the previous line `return programRepository.findAll()` would be sufficient.

Reply:
also, by removing the ProgramSpecificationBuilder, calling `.getRegions` or any other collection, on a program will result in a LazyInitializationException, since it was not fetched. Verify how listPrograms is used downstream to ensure that not fetching join entities is ok.

Reply:
`ProgramSpecificationBuilder.setupFetchStrategy` is the reason listPrograms shows duplicates programs because the generated sql is: `program_service=# select (id, short_name) from program left outer join program_cancer on (program.id = program_cancer.program_id) left outer join program_institution on (program.id = program_institution.program_id) left outer join program_primary_site on (program.id = program_primary_site.program_id) left outer join program_country on (program.id = program_country.program_id) left outer join program_region on (program.id = program_region.program_id);`.

By using `@Transcational` on listPrograms, all associations will be fetched without getting lazyInitializationException.

Reply:
Without the use of the SpecificationBuilder, the N+1 Query problem can arise. Although its convenient, its not efficient since you will be doing N selects as opposed to 1 join. The duplicates can be removed by adding the distinct keyword
https://github.com/overture-stack/ego/blob/develop/src/main/java/bio/overture/ego/repository/queryspecification/UserSpecification.java#L49
to all the methods in [AbstractSpecificationBuilder](https://github.com/overture-stack/ego/blob/develop/src/main/java/bio/overture/ego/repository/queryspecification/builder/AbstractSpecificationBuilder.java)


In addition, replace line 306 (program stream) with line 305 (findall)